app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -64,12 +66,12 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(hawk_escape_id(feed)).properties['Machine Id']
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
-        os = os_for(feed)
+        os = os_for(hawk_escape_id(feed))
```

Comment:
trick? not sure what do you mean. if cross linking doesn't work then
there's a bug and it should be fixed.

On Tue, Aug 2, 2016 at 3:24 PM, Juraci Paixão Kröhling <
notifications@github.com> wrote:

> In app/models/manageiq/providers/hawkular/middleware_manager.rb
> <https://github.com/ManageIQ/manageiq/pull/10048#discussion_r73144495>:
>
> >      end
> >
> >      def os_resource_for(feed)
> >        with_provider_connection do |connection|
> > -        os = os_for(feed)
> > +        os = os_for(hawk_escape_id(feed))
>
> To verify on the UI, you'll also need "the other side" of the link, ie, a
> RHEV-M instance. We have both on torii, so, you could try adding them on
> your local MiQ instance and see if it still works. I remember theute
> mentioning that there was a trick to get it working properly. Perhaps
> @abonas <https://github.com/abonas> would know about such trick?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/10048/files/721c2f6500c7146ab25c43fb6474c328a117ca9f#r73144495>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF_IfeomXr7P47ZLwRIOYd5YEx984oTSks5qbzbpgaJpZM4JUqzX>
> .
>